src/App.js

```diff
@@ -28,31 +28,100 @@ const generateSquares = () => {
 const App = () => {
 
   const [squares, setSquares] = useState(generateSquares());
-
+  const [isXTurn, setXTurn] = useState(true);
+  const [winner, setWinner] = useState(null);
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
+ 
   // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+
+  const onClickCallback = ( (squareId) => {
+
+    // will not continue the game if there is already a winner
+    if (winner) return;
+
+    let copiedSquares = [...squares];
+    for(let row of copiedSquares) {
+      for(let square of row) {
+       if (square.id === squareId && square.value === '') {
+          // assign square.value base on the current valuse of isXTurn
+          square.value = isXTurn ? PLAYER_1 : PLAYER_2;
+          setXTurn(!isXTurn);
+          isXTurn ? setCurrentPlayer(PLAYER_2) : setCurrentPlayer(PLAYER_1);
+        } 
+      }
+    }
+    setSquares(copiedSquares);
+    checkForWinner(copiedSquares);
+  }) 
 
 
   const checkForWinner = () => {
     // Complete in Wave 3
+    // check rows
```

Comment:
I appreciate that you added these comments to help the reader follow what the code is doing.
Often when you find yourself writing comments like this to describe what a block of code is doing, it's a good cue that you could make that block of code it's own function to further improve readability.